src/cow-react/modules/trade/hooks/useOnCurrencySelection.ts

```diff
@@ -24,22 +26,31 @@ function useResolveCurrencyAddressOrSymbol(): (currency: Currency | null) => str
  * if there are more than one token with the same symbol
  * @see useResetStateWithSymbolDuplication.ts
  */
-export function useOnCurrencySelection(): (field: Field, currency: Currency) => void {
+export function useOnCurrencySelection(): CurrencySelectionCallback {
   const { chainId } = useWeb3React()
-  const { inputCurrency, outputCurrency } = useLimitOrdersTradeState()
-  const limitOrdersNavigate = useLimitOrdersNavigate()
+  const tradeState = useTradeState()
+  const navigate = useTradeNavigate()
   const resolveCurrencyAddressOrSymbol = useResolveCurrencyAddressOrSymbol()
 
   return useCallback(
-    (field: Field, currency: Currency) => {
+    (field: Field, currency: Currency | null) => {
```

Comment:
Nitpick: can't you reuse the type from `CurrencySelectionCallback`?

Reply:
This is function body